### PR TITLE
feat: add support for public IP

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Add support for Public IP connections to AlloyDB instances via Go Connector. Default  option for `ipType` will be Private IP but users wishing to use Public IP can use the new `WithPublicIP()` DialOption.

PR Checklist:
- [X] use v1alpha client
- [X] allow multiple `ipAddr` to be returned
- [x] update existing unit tests
- [x] add new integration test
- [x] Update README with sample

### CODE REVIEW

File: e2e_test.go [hunk 1]
```diff
@@ -227,3 +227,40 @@ func TestAutoIAMAuthN(t *testing.T) {
 	}
 	t.Log(tt)
 }
+
+func TestPublicIP(t *testing.T) {
```
Comment (enocom) [inline e2e_test.go:231]: Shall we put region tags around this usage so we can get it into the public docs?

**Review by enocom — APPROVED** [review]
LGTM. We'll merge this once the feature is fully rolled out.

File: e2e_test.go (hunk 1 above)
Comment (jackwotherspoon) [reply e2e_test.go:231]: Do we want to do that as a follow-up PR and arrange both Connector Public IP and Connector Auto IAM AuthN like our direct path tests?

Comment (jackwotherspoon) [reply e2e_test.go:231]: Because we don't currently have region tags around Connector Auto IAM AuthN...

**Pull request merged**